packages/gatsby/src/schema/schema.js

```diff
@@ -72,12 +73,17 @@ const updateSchemaComposer = async ({
   schemaComposer.addDirective(InferDirective)
   schemaComposer.addDirective(DontInferDirective)
   await addTypeDefs({ schemaComposer, parentSpan, typeDefs })
-  await addInferredTypes({ schemaComposer, nodeStore, parentSpan })
-  await addSetFieldsOnGraphQLNodeTypeFields({
+  nodeStore.getTypes().forEach(typeName => {
```

Comment:
This is so people don't have to use `implements Node` in addTypeDefs, right?

Reply:
It's so that types exist when `setFieldsOnGraphQLNodeTypeFields` is called.

Reply:
`implements Node` is mandatory.